streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -38,7 +38,10 @@ private Sensors() {}
 
     public static Sensor lateRecordDropSensor(final InternalProcessorContext context) {
         final StreamsMetricsImpl metrics = context.metrics();
+        final String threadId = Thread.currentThread().getName();
```

Comment:
Is this always right? It would set the threadId as the name of the thread that constructed the sensor, but is this always the StreamThread? Even if it is today, it wouldn't be outrageous to think in the future that another thread might build a task and then pass it to the thread to execute. Such a refactoring would break this logic, but it would be very subtle.

In contrast, the old code here explicitly passed the thread name via the context, so it would never be set incorrectly just by changing the way that the code is invoked. Perhaps we could preserve this property, for example by passing the thread name as a method argument.

Reply:
Just checked the code path, passing the thread name via `GlobalStreamThread / StreamThread -> StreamTask -> ProcessorContextImpl` maybe fine.

Reply:
Ok, sounds fine to me, as long as we're aware of the risk.